package.json

```diff
@@ -12,7 +12,7 @@
   ],
   "scripts": {
     "build": "yarn build-bundles && cross-env NODE_ENV=production PARCEL_BUILD_ENV=production gulp",
-    "build-bundles": "rimraf --glob packages/*/*/lib && cross-env NODE_ENV=production PARCEL_BUILD_ENV=production PARCEL_SELF_BUILD=true parcel build packages/core/{fs,codeframe,package-manager,utils} packages/reporters/{cli,dev-server} packages/utils/{parcel-lsp,parcel-lsp-protocol}",
+    "build-bundles": "rimraf --glob packages/*/*/lib && lerna run dev:prepare && cross-env NODE_ENV=production PARCEL_BUILD_ENV=production PARCEL_SELF_BUILD=true parcel build packages/core/{fs,codeframe,package-manager,utils} packages/reporters/{cli,dev-server} packages/utils/{parcel-lsp,parcel-lsp-protocol}",
```

Comment:
Were you going to ensure that `dev:prepare` does not run during the release? It gets triggered via `prepublishOnly` FYI

Reply:
This is fine & required as long as `build` will run afterwards.

In order for parcel to self-build, the production build sequence needs to be:

* Link `node_modules/.bin/parcel -> packages/core/parcel/bin/parcel.js` ; this needs to point at the unbuilt file at this point
* `yarn build-bundles` go and run parcel over `@parcel/fs` and such; parcel needs to be pointed at an unbuilt file, because if it pointed at a pre-built dir. we would have a task cycle
* now `yarn build / gulp` will run and all the `lib` directories written in `dev:prepare` should be overwritten. 

We can add extra validation to make sure we don't publish the wrong assets though.

Reply:
Yeah, it's more about making sure the files get overwritten. Otherwise the build should fail imo